src/Components/Search/Search.js

```diff
@@ -31,6 +51,8 @@ const Search = ({addRecipes}) => {
           ><img src={searchIcon} alt="search button" className="search-btn-img"/>
```

Comment:
Instead of checking for errors on submitSearch, you could check for input errors every time the state changes. You could then have the button disabled until that input error checking fn returns true/false/however you set it up. Not sure if that makes sense, but it's what we did on freshly fetched to validate the zipcode. In this instance, where all you are really checking for is whether the search is '', there might be a more simple way to do it... like maybe the button is disabled as long as search does not have length, or something like that.

Reply:
That makes sense. Some of the error is dependent on what comes back from the search so I'd have to trigger it on the submit. I'll play around with disabling the button based on the search state being '', that is a great idea- Why handle for an error that can be easily preventable?